### PR TITLE
resource_control: support unlimited keyword when setting the resource group

### LINKED ISSUE #54413: support unlimit keyworld when setting the resource group
## Feature Request

**Is your feature request related to a problem? Please describe:**
```
CREATE RESOURCE GROUP IF NOT EXISTS rg1
  RU_PER_SEC = 100
  PRIORITY = HIGH
  BURSTABLE;
```

**Describe the feature you'd like:**

supports setting `RU_PER_SEC = unlimited`, like the `default` group settings.

and also `select * from information_schem.resource_groups`  list will shown as `unlimit`

### PR DESCRIPTION
**Pull request by HuSharp**
### What problem does this PR solve?

Issue Number: close #54413

Problem Summary:

### What changed and how does it work?

```bash
mysql> SELECT * FROM information_schema.resource_groups;
+---------+------------+----------+-----------+-------------+------------+
| NAME    | RU_PER_SEC | PRIORITY | BURSTABLE | QUERY_LIMIT | BACKGROUND |
+---------+------------+----------+-----------+-------------+------------+
| default | UNLIMITED  | MEDIUM   | YES       | NULL        | NULL       |
+---------+------------+----------+-----------+-------------+------------+
1 row in set (0.00 sec)

mysql> CREATE RESOURCE GROUP IF NOT EXISTS rg1   RU_PER_SEC = 100   BURSTABLE;
Query OK, 0 rows affected (0.07 sec)

mysql> CREATE RESOURCE GROUP IF NOT EXISTS rg2   RU_PER_SEC = unlimited   BURSTABLE;
Query OK, 0 rows affected (0.06 sec)

mysql> CREATE RESOURCE GROUP IF NOT EXISTS rg3   RU_PER_SEC = UNLIMITED   BURSTABLE;
Query OK, 0 rows affected (0.07 sec)

mysql> SELECT * FROM information_schema.resource_groups;
+---------+------------+----------+-----------+-------------+------------+
| NAME    | RU_PER_SEC | PRIORITY | BURSTABLE | QUERY_LIMIT | BACKGROUND |
+---------+------------+----------+-----------+-------------+------------+
| default | UNLIMITED  | MEDIUM   | YES       | NULL        | NULL       |
| rg1     | 100        | MEDIUM   | YES       | NULL        | NULL       |
| rg2     | UNLIMITED  | MEDIUM   | YES       | NULL        | NULL       |
| rg3     | UNLIMITED  | MEDIUM   | YES       | NULL        | NULL       |
+---------+------------+----------+-----------+-------------+------------+
4 rows in set (0.00 sec)

mysql> ALTER RESOURCE GROUP rg2 RU_PER_SEC = 200;
Query OK, 0 rows affected (0.05 sec)

mysql> SELECT * FROM information_schema.resource_groups;
+---------+------------+----------+-----------+-------------+------------+
| NAME    | RU_PER_SEC | PRIORITY | BURSTABLE | QUERY_LIMIT | BACKGROUND |
+---------+------------+----------+-----------+-------------+------------+
| default | UNLIMITED  | MEDIUM   | YES       | NULL        | NULL       |
| rg1     | 100        | MEDIUM   | YES       | NULL        | NULL       |
| rg2     | 200        | MEDIUM   | YES       | NULL        | NULL       |
| rg3     | UNLIMITED  | MEDIUM   | YES       | NULL        | NULL       |
+---------+------------+----------+-----------+-------------+------------+
4 rows in set (0.00 sec)

mysql> CREATE RESOURCE GROUP IF NOT EXISTS rg4   RU_PER_SEC = UNLIMITED   BURSTABLE=FALSE;
Query OK, 0 rows affected (0.06 sec)

mysql> SELECT * FROM information_schema.resource_groups;
+---------+------------+----------+-----------+-------------+------------+
| NAME    | RU_PER_SEC | PRIORITY | BURSTABLE | QUERY_LIMIT | BACKGROUND |
+---------+------------+----------+-----------+-------------+------------+
| default | UNLIMITED  | MEDIUM   | YES       | NULL        | NULL       |
| rg1     | 100        | MEDIUM   | YES       | NULL        | NULL       |
| rg2     | 200        | MEDIUM   | YES       | NULL        | NULL       |
| rg3     | UNLIMITED  | MEDIUM   | YES       | NULL        | NULL       |
| rg4     | UNLIMITED  | MEDIUM   | NO        | NULL        | NULL       |
+---------+------------+----------+-----------+-------------+------------+
5 rows in set (0.01 sec)

mysql> CREATE RESOURCE GROUP IF NOT EXISTS rg5   RU_PER_SEC = 'check'   BURSTABLE=FALSE;
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use line 1 column 74 near "BURSTABLE=FALSE"The RU_PER_SEC value 'check' is not valid.

mysql> alter resource group rg1 RU_PER_SEC= unlimited QUERY_LIMIT=(EXEC_ELAPSED='15s' ACTION DRYRUN WATCH SIMILAR DURATION '10m0s')
    -> ;
Query OK, 0 rows affected (0.06 sec)

mysql> SELECT * FROM information_schema.resource_groups;
+---------+------------+----------+-----------+-------------------------------------------------------------------+------------+
| NAME    | RU_PER_SEC | PRIORITY | BURSTABLE | QUERY_LIMIT                                                       | BACKGROUND |
+---------+------------+----------+-----------+-------------------------------------------------------------------+------------+
| default | UNLIMITED  | MEDIUM   | YES       | NULL                                                              | NULL       |
| rg1     | UNLIMITED  | MEDIUM   | YES       | EXEC_ELAPSED='15s', ACTION=DRYRUN, WATCH=SIMILAR DURATION='10m0s' | NULL       |
| rg2     | 200        | MEDIUM   | YES       | NULL                                                              | NULL       |
| rg3     | UNLIMITED  | MEDIUM   | YES       | NULL                                                              | NULL       |
| rg4     | UNLIMITED  | MEDIUM   | NO        | NULL                                                              | NULL       |
+---------+------------+----------+-----------+-------------------------------------------------------------------+------------+
5 rows in set (0.01 sec)
```

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by HuSharp** [comment]
@nolouch @JmPotato PTAL, thx!

File: pkg/parser/keywords.go [hunk 1]
```diff
@@ -622,6 +622,7 @@ var Keywords = []KeywordsType{
 	{"VALIDATION", false, "unreserved"},
 	{"VALUE", false, "unreserved"},
 	{"VARIABLES", false, "unreserved"},
+	{"VECTOR", false, "unreserved"},
```
Comment (HuSharp) [inline pkg/parser/keywords.go:625]: Someone who forgot to update, maybe we need a lint

File: pkg/parser/parser.y [hunk 2]
```diff
@@ -1866,6 +1866,17 @@ DirectResourceGroupOption:
 	{
 		$$ = &ast.ResourceGroupOption{Tp: ast.ResourceRURate, UintValue: $3.(uint64)}
 	}
+|	"RU_PER_SEC" EqOpt stringLit
```
Comment (nolouch) [inline pkg/parser/parser.y:1869]: Can we directly use "UNLIMITED"? like `WatchDurationOption`

Comment (HuSharp) [reply pkg/parser/parser.y:1869]: `WatchDurationOption` still needs to be checked by the `strings.ToLower`, which is maybe no different than now. 

https://github.com/pingcap/tidb/blob/8f0c053a38b4b99d00d5eb6f466c783d11a1f899/pkg/parser/parser.y#L1834-L1849

Comment (HuSharp) [reply pkg/parser/parser.y]: good catch, using `UNLIMITED` now.

File: pkg/parser/keywords.go (hunk 1 above)
Comment (HuSharp) [reply pkg/parser/keywords.go:625]: added https://github.com/pingcap/tidb/pull/54827

**Review by nolouch — APPROVED** [review]

File: pkg/parser/ast/ddl.go [hunk 3]
```diff
@@ -2176,6 +2176,7 @@ type ResourceGroupOption struct {
 	StrValue          string
 	UintValue         uint64
 	BoolValue         bool
+	RUUnlimited       bool // mark `RU_PER_SEC` is `unlimited`
```
Comment (glorv) [inline pkg/parser/ast/ddl.go:2179]: What about share BoolValue?

**Review by glorv — COMMENTED** [review]
rest LGTM

File: pkg/parser/ast/ddl.go (hunk 3 above)
Comment (HuSharp) [reply pkg/parser/ast/ddl.go:2179]: Okay, It does work, the fields look like all mixed up

**Review by glorv — APPROVED** [review]

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Review by tangenta — APPROVED** [review]

**Review by songrijie — APPROVED** [review]

**Review by easonn7 — APPROVED** [review]

**Pull request merged**